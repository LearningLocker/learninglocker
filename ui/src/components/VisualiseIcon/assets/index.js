import SESSIONS_GREY_IMAGE from './ll-icon-accumulation-grey.svg';
import SESSIONS_IMAGE from './ll-icon-accumulation.svg';
import LEADERBOARD_GREY_IMAGE from './ll-icon-bar-graph-grey.svg';
import LEADERBOARD_IMAGE from './ll-icon-bar-graph.svg';
import BIAXIAL_GREY_IMAGE from './ll-icon-column-and-frequency-grey.svg';
import STATEMENTS_GREY_IMAGE from './ll-icon-column-graph-grey.svg';
import STATEMENTS_IMAGE from './ll-icon-column-graph.svg';
import XVSY_GREY_IMAGE from './ll-icon-correlation-grey.svg';
import XVSY_IMAGE from './ll-icon-correlation.svg';
import COUNTER_GREY_IMAGE from './ll-icon-counter-grey.svg';
import COUNTER_IMAGE from './ll-icon-counter.svg';
import FREQUENCY_GREY_IMAGE from './ll-icon-frequency-grey.svg';
import FREQUENCY_IMAGE from './ll-icon-frequency.svg';
import PIE_GREY_IMAGE from './ll-icon-pie-chart-grey.svg';
import PIE_IMAGE from './ll-icon-pie-chart.svg';
import TABLE_AND_STATEMENT_GREY_IMAGE from './ll-icon-table-and-statement-grey.svg';
import TABLE_GREY_IMAGE from './ll-icon-table-grey.svg';
import TABLE_IMAGE from './ll-icon-table.svg';

export {
  SESSIONS_GREY_IMAGE,
  SESSIONS_IMAGE,
  LEADERBOARD_GREY_IMAGE,
  LEADERBOARD_IMAGE,
  BIAXIAL_GREY_IMAGE,
  STATEMENTS_GREY_IMAGE,
  STATEMENTS_IMAGE,
  XVSY_GREY_IMAGE,
  XVSY_IMAGE,
  COUNTER_GREY_IMAGE,
  COUNTER_IMAGE,
  FREQUENCY_GREY_IMAGE,
  FREQUENCY_IMAGE,
  PIE_GREY_IMAGE,
  PIE_IMAGE,
  TABLE_AND_STATEMENT_GREY_IMAGE,
  TABLE_GREY_IMAGE,
  TABLE_IMAGE,
};
