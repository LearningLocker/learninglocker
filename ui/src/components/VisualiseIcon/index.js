/* eslint-disable react/jsx-indent */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  LEADERBOARD,
  XVSY,
  STATEMENTS,
  FREQUENCY,
  COUNTER,
  PIE,
  TEMPLATE_ACTIVITY_OVER_TIME,
  TEMPLATE_LAST_7_DAYS_STATEMENTS,
  TEMPLATE_MOST_ACTIVE_PEOPLE,
  TEMPLATE_MOST_POPULAR_ACTIVITIES,
  TEMPLATE_MOST_POPULAR_VERBS,
  TEMPLATE_WEEKDAYS_ACTIVITY,
  TEMPLATE_STREAM_INTERACTIONS_VS_ENGAGEMENT,
  TEMPLATE_STREAM_COMMENT_COUNT,
  TEMPLATE_STREAM_LEARNER_INTERACTIONS_BY_DATE_AND_VERB,
  TEMPLATE_STREAM_USER_ENGAGEMENT_LEADERBOARD,
  TEMPLATE_STREAM_PROPORTION_OF_SOCIAL_INTERACTIONS,
  TEMPLATE_STREAM_ACTIVITIES_WITH_MOST_COMMENTS,
  TEMPLATE_LEARNING_EXPERIENCE_TYPE,
} from 'lib/constants/visualise';
import {
  LEADERBOARD_GREY_IMAGE,
  LEADERBOARD_IMAGE,
  STATEMENTS_GREY_IMAGE,
  STATEMENTS_IMAGE,
  XVSY_GREY_IMAGE,
  XVSY_IMAGE,
  COUNTER_GREY_IMAGE,
  COUNTER_IMAGE,
  FREQUENCY_GREY_IMAGE,
  FREQUENCY_IMAGE,
  PIE_GREY_IMAGE,
  PIE_IMAGE,
  TABLE_GREY_IMAGE,
  TABLE_IMAGE,
} from './assets';

const Image = styled.img`
  height: ${props => props.isSmall && '30px' || 'initial'}
`;

/**
 * @param {string} type
 * @returns {string}
 */
const getTitle = (type) => {
  switch (type) {
    case LEADERBOARD: return 'Bar';
    case XVSY: return 'Correlation';
    case STATEMENTS: return 'Column';
    case FREQUENCY: return 'Line';
    case COUNTER: return 'Counter';
    case PIE: return 'Pie';
    default: return '';
  }
};

/**
 * @param {string} type
 * @param {boolean} sourceView
 * @returns {string}
 */
const getImageSrc = (type, sourceView) => {
  if (sourceView) {
    switch (type) {
      case LEADERBOARD:
      case XVSY:
      case STATEMENTS:
      case FREQUENCY:
      case COUNTER:
      case PIE:
        return TABLE_IMAGE;
      default:
        return TABLE_GREY_IMAGE;
    }
  }

  switch (type) {
    case LEADERBOARD:
      return LEADERBOARD_IMAGE;
    case TEMPLATE_MOST_ACTIVE_PEOPLE:
    case TEMPLATE_MOST_POPULAR_ACTIVITIES:
    case TEMPLATE_MOST_POPULAR_VERBS:
    case TEMPLATE_STREAM_USER_ENGAGEMENT_LEADERBOARD:
    case TEMPLATE_STREAM_ACTIVITIES_WITH_MOST_COMMENTS:
      return LEADERBOARD_GREY_IMAGE;
    case TEMPLATE_LEARNING_EXPERIENCE_TYPE:
      return LEADERBOARD_GREY_IMAGE;
    case XVSY:
      return XVSY_IMAGE;
    case TEMPLATE_STREAM_INTERACTIONS_VS_ENGAGEMENT:
      return XVSY_GREY_IMAGE;
    case STATEMENTS:
      return STATEMENTS_IMAGE;
    case TEMPLATE_WEEKDAYS_ACTIVITY:
    case TEMPLATE_STREAM_LEARNER_INTERACTIONS_BY_DATE_AND_VERB:
      return STATEMENTS_GREY_IMAGE;
    case FREQUENCY:
      return FREQUENCY_IMAGE;
    case TEMPLATE_ACTIVITY_OVER_TIME:
      return FREQUENCY_GREY_IMAGE;
    case COUNTER:
      return COUNTER_IMAGE;
    case TEMPLATE_LAST_7_DAYS_STATEMENTS:
    case TEMPLATE_STREAM_COMMENT_COUNT:
      return COUNTER_GREY_IMAGE;
    case PIE:
      return PIE_IMAGE;
    case TEMPLATE_STREAM_PROPORTION_OF_SOCIAL_INTERACTIONS:
      return PIE_GREY_IMAGE;
    default:
      return '';
  }
};

const VisualiseIcon = ({
  type,
  sourceView,
  isSmall = true, // // [Viz Refactor] TODO: Remove this property
}) => {
  const src = getImageSrc(type, sourceView);
  if (src === '') {
    return null;
  }

  // TODO: alt should be mapped from image src, not type
  return (
    <Image
      isSmall={isSmall}
      src={src}
      alt={getTitle(type)} />
  );
};

VisualiseIcon.propTypes = {
  sourceView: PropTypes.bool.isRequired,
  type: PropTypes.string,
  isSmall: PropTypes.bool,
};

export default React.memo(VisualiseIcon);
