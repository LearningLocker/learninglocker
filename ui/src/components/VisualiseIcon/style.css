.visualisationIcon {
  text-align: center;
  vertical-align: top;
  display: inline-block;
  margin: 3px;
  width: 109px;
}

.visualisationIcon h5 {
  text-align: center;
  padding: 0px 4px;
  font-size: 12px;
  font-weight: bold;
  margin-top: 0;
}

.visualisationIcon.active {
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.3);
}

.visualisationIcon img {
  width: 100%;
}

.visualisationSmall {
  height: 30px;
}
