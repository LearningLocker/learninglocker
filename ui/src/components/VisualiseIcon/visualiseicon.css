.visualisation-icon {
  text-align: center;
  vertical-align: top;
  display: inline-block;
  margin: 3px;
  width: 109px; }
.visualisation-icon h5 {
  text-align: center;
  padding: 0px 4px;
  font-size: 12px;
  font-weight: bold;
  margin-top: 0; }
.visualisation-icon.active {
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.3); }
.visualisation-icon img {
  width: 100%; }


.visualisationSmall {
  height: 30px;
}