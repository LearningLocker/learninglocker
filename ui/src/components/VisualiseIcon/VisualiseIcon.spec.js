import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import VisualiseIcon from './index';

describe('VisualiseIcon', () => {
  it('should render small icon', () => {
    const result = renderer
      .create(
        <VisualiseIcon
          type="XVSY"
          sourceView={false} />
      )
      .toJSON();

    expect(result).toMatchSnapshot();
    expect(result).toHaveStyleRule('height', '30px');
  });
});
