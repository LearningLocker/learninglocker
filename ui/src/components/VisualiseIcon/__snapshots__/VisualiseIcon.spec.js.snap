// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualiseIcon should render small icon 1`] = `
.c0 {
  height: 30px;
}

<img
  alt="Correlation"
  className="c0"
  src="test-file-stub"
/>
`;
